packages/mobile/src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx

```diff
@@ -112,8 +121,21 @@ export const AddToPlaylistDrawer = () => {
               toast({ content: messages.hiddenAdd })
               return
             }
-            toast({ content: messages.addedToast })
-            dispatch(addTrackToPlaylist(trackId!, item.playlist_id))
+
+            const doesPlaylistContainTrack =
+              playlistTrackIdMap[item.playlist_id]?.includes(trackId)
+
+            if (isPlaylistUpdatesEnabled && doesPlaylistContainTrack) {
+              dispatch(
+                openDuplicateAddConfirmation({
+                  playlistId: item.playlist_id,
+                  trackId: trackId!
+                })
+              )
+            } else {
+              toast({ content: messages.addedToast })
+              dispatch(addTrackToPlaylist(trackId!, item.playlist_id))
```

Comment:
qq regarding `!` usage vs an `if ()` check? i think in either case it's not really possible to hit, but ive personally done the `if (trackId)` check usually